torch/_utils.py

```diff
@@ -326,3 +328,42 @@ def dec(fun):
         fun.__annotations__['return'] = ret
         return fun
     return dec
+
+
+# NOTE [ Python Traceback Reference Cycle Problem ]
+#
+# When using sys.exc_info(), it is important to **not** store the exc_info[2],
+# which is the traceback, because otherwise you will run into the traceback
+# reference cycle problem, i.e., the traceback holding reference to the frame,
+# and the frame (which holds reference to all the object in its temporary scope)
+# holding reference the traceback.
+
+class KeyErrorMessage(str):
+    r"""str subclass that returns itself in repr"""
+    def __repr__(self):
+        return self
+
+
+class ExceptionWrapper(object):
+    r"""Wraps an exception plus traceback to communicate across threads"""
+    def __init__(self, exc_info=None, where="in background"):
```

Comment:
why is `exc_info` made optional here?

Reply:
The existing code just passed `sys.exc_info()` to the exception wrapper. I moved this into the constructor, but still allow overriding it (just in case, not sure if there is a good use case for it). Would you prefer removing the argument altogether? Or should we extend the docstring?

Reply:
This is fine. Thanks for explanation!